diff --git a/modules/photo/src/inpaint.cpp b/modules/photo/src/inpaint.cpp
index d9a7dbc58b..b436168212 100644
--- a/modules/photo/src/inpaint.cpp
+++ b/modules/photo/src/inpaint.cpp
@@ -45,9 +45,6 @@
 //
 // */
 
-#include <queue>
-#include <type_traits>
-
 #include "precomp.hpp"
 #include "opencv2/imgproc/imgproc_c.h"
 #include "opencv2/photo/legacy/constants_c.h"
@@ -56,16 +53,6 @@
 #define CV_MAT_ELEM_PTR_FAST( mat, row, col, pix_size )  \
      ((mat).data.ptr + (size_t)(mat).step*(row) + (pix_size)*(col))
 
-template<typename T>
-typename std::enable_if<std::is_floating_point<T>::value, T>::type round_cast(float val) {
-   return cv::saturate_cast<T>(val);
-}
-
-template<typename T>
-typename std::enable_if<!std::is_floating_point<T>::value, T>::type round_cast(float val) {
-   return cv::saturate_cast<T>(val + 0.5);
-}
-
 inline float
 min4( float a, float b, float c, float d )
 {
@@ -84,16 +71,8 @@ typedef struct CvHeapElem
 {
     float T;
     int i,j;
-    int order;  // to keep insertion order
-
-    bool operator > (const CvHeapElem& rhs) const {
-        if (T > rhs.T) {
-            return true;
-        } else if (T < rhs.T) {
-            return false;
-        }
-        return order > rhs.order;
-    }
+    struct CvHeapElem* prev;
+    struct CvHeapElem* next;
 }
 CvHeapElem;
 
@@ -105,10 +84,42 @@ private:
     CvPriorityQueueFloat& operator=(const CvPriorityQueueFloat &); // assign disabled
 
 protected:
-    std::priority_queue<CvHeapElem, std::vector<CvHeapElem>,std::greater<CvHeapElem> > queue;
-    int next_order;
+    CvHeapElem *mem,*empty,*head,*tail;
+    int num,in;
 
 public:
+    bool Init( const CvMat* f )
+    {
+        int i,j;
+        for( i = num = 0; i < f->rows; i++ )
+        {
+            for( j = 0; j < f->cols; j++ )
+                num += CV_MAT_ELEM(*f,uchar,i,j)!=0;
+        }
+        if (num<=0) return false;
+        mem = (CvHeapElem*)cvAlloc((num+2)*sizeof(CvHeapElem));
+        if (mem==NULL) return false;
+
+        head       = mem;
+        head->i    = head->j = -1;
+        head->prev = NULL;
+        head->next = mem+1;
+        head->T    = -FLT_MAX;
+        empty      = mem+1;
+        for (i=1; i<=num; i++) {
+            mem[i].prev   = mem+i-1;
+            mem[i].next   = mem+i+1;
+            mem[i].i      = -1;
+            mem[i].T      = FLT_MAX;
+        }
+        tail       = mem+i;
+        tail->i    = tail->j = -1;
+        tail->prev = mem+i-1;
+        tail->next = NULL;
+        tail->T    = FLT_MAX;
+        return true;
+    }
+
     bool Add(const CvMat* f) {
         int i,j;
         for (i=0; i<f->rows; i++) {
@@ -122,33 +133,71 @@ public:
     }
 
     bool Push(int i, int j, float T) {
-        queue.push({T, i, j, next_order});
-        ++next_order;
+        CvHeapElem *tmp=empty,*add=empty;
+        if (empty==tail) return false;
+        while (tmp->prev->T>T) tmp = tmp->prev;
+        if (tmp!=empty) {
+            add->prev->next = add->next;
+            add->next->prev = add->prev;
+            empty = add->next;
+            add->prev = tmp->prev;
+            add->next = tmp;
+            add->prev->next = add;
+            add->next->prev = add;
+        } else {
+            empty = empty->next;
+        }
+        add->i = i;
+        add->j = j;
+        add->T = T;
+        in++;
+        //      printf("push i %3d  j %3d  T %12.4e  in %4d\n",i,j,T,in);
         return true;
     }
 
     bool Pop(int *i, int *j) {
-        if (queue.empty()) {
-            return false;
-        }
-        *i = queue.top().i;
-        *j = queue.top().j;
-        queue.pop();
+        CvHeapElem *tmp=head->next;
+        if (empty==tmp) return false;
+        *i = tmp->i;
+        *j = tmp->j;
+        tmp->prev->next = tmp->next;
+        tmp->next->prev = tmp->prev;
+        tmp->prev = empty->prev;
+        tmp->next = empty;
+        tmp->prev->next = tmp;
+        tmp->next->prev = tmp;
+        empty = tmp;
+        in--;
+        //      printf("pop  i %3d  j %3d  T %12.4e  in %4d\n",tmp->i,tmp->j,tmp->T,in);
         return true;
     }
 
     bool Pop(int *i, int *j, float *T) {
-        if (queue.empty()) {
-            return false;
-        }
-        *i = queue.top().i;
-        *j = queue.top().j;
-        *T = queue.top().T;
-        queue.pop();
+        CvHeapElem *tmp=head->next;
+        if (empty==tmp) return false;
+        *i = tmp->i;
+        *j = tmp->j;
+        *T = tmp->T;
+        tmp->prev->next = tmp->next;
+        tmp->next->prev = tmp->prev;
+        tmp->prev = empty->prev;
+        tmp->next = empty;
+        tmp->prev->next = tmp;
+        tmp->next->prev = tmp;
+        empty = tmp;
+        in--;
+        //      printf("pop  i %3d  j %3d  T %12.4e  in %4d\n",tmp->i,tmp->j,tmp->T,in);
         return true;
     }
 
-    CvPriorityQueueFloat(void) : queue(), next_order() {
+    CvPriorityQueueFloat(void) {
+        num=in=0;
+        mem=empty=head=tail=NULL;
+    }
+
+    ~CvPriorityQueueFloat(void)
+    {
+        cvFree( &mem );
     }
 };
 
@@ -340,7 +389,7 @@ icvTeleaInpaintFMM(const CvMat *f, CvMat *t, CvMat *out, int range, CvPriorityQu
                                     gradI.y=0;
                                  }
                               }
-                              Ia[color] += (float)w * (float)(CV_MAT_3COLOR_ELEM(*out,uchar,k-1,l-1,color));
+                              Ia[color] += (float)w * (float)(CV_MAT_3COLOR_ELEM(*out,uchar,km,lm,color));
                               Jx[color] -= (float)w * (float)(gradI.x*r.x);
                               Jy[color] -= (float)w * (float)(gradI.y*r.y);
                               s[color]  += w;
@@ -350,8 +399,8 @@ icvTeleaInpaintFMM(const CvMat *f, CvMat *t, CvMat *out, int range, CvPriorityQu
                   }
                }
                for (color=0; color<=2; color++) {
-                  sat = (float)(Ia[color]/s[color]+(Jx[color]+Jy[color])/(sqrt(Jx[color]*Jx[color]+Jy[color]*Jy[color])+1.0e-20f));
-                  CV_MAT_3COLOR_ELEM(*out,uchar,i-1,j-1,color) = round_cast<uchar>(sat);
+                  sat = (float)((Ia[color]/s[color]+(Jx[color]+Jy[color])/(sqrt(Jx[color]*Jx[color]+Jy[color]*Jy[color])+1.0e-20f)+0.5f));
+                  CV_MAT_3COLOR_ELEM(*out,uchar,i-1,j-1,color) = cv::saturate_cast<uchar>(sat);
                }
 
                CV_MAT_ELEM(*f,uchar,i,j) = BAND;
@@ -452,7 +501,7 @@ icvTeleaInpaintFMM(const CvMat *f, CvMat *t, CvMat *out, int range, CvPriorityQu
                                     gradI.y=0;
                                  }
                               }
-                              Ia += (float)w * (float)(CV_MAT_ELEM(*out,data_type,k-1,l-1));
+                              Ia += (float)w * (float)(CV_MAT_ELEM(*out,data_type,km,lm));
                               Jx -= (float)w * (float)(gradI.x*r.x);
                               Jy -= (float)w * (float)(gradI.y*r.y);
                               s  += w;
@@ -460,9 +509,9 @@ icvTeleaInpaintFMM(const CvMat *f, CvMat *t, CvMat *out, int range, CvPriorityQu
                         }
                      }
                   }
-                  sat = (float)(Ia/s+(Jx+Jy)/(sqrt(Jx*Jx+Jy*Jy)+1.0e-20f));
+                  sat = (float)((Ia/s+(Jx+Jy)/(sqrt(Jx*Jx+Jy*Jy)+1.0e-20f)+0.5f));
                   {
-                  CV_MAT_ELEM(*out,data_type,i-1,j-1) = round_cast<data_type>(sat);
+                  CV_MAT_ELEM(*out,data_type,i-1,j-1) = cv::saturate_cast<data_type>(sat);
                   }
                }
 
@@ -555,7 +604,7 @@ icvNSInpaintFMM(const CvMat *f, CvMat *t, CvMat *out, int range, CvPriorityQueue
                                  dir = (float)fabs(VectorScalMult(r,gradI)/sqrt(VectorLength(r)*VectorLength(gradI)));
                               }
                               w = dst*dir;
-                              Ia[color] += (float)w * (float)(CV_MAT_3COLOR_ELEM(*out,uchar,k-1,l-1,color));
+                              Ia[color] += (float)w * (float)(CV_MAT_3COLOR_ELEM(*out,uchar,km,lm,color));
                               s[color]  += w;
                            }
                         }
@@ -645,7 +694,7 @@ icvNSInpaintFMM(const CvMat *f, CvMat *t, CvMat *out, int range, CvPriorityQueue
                                  dir = (float)fabs(VectorScalMult(r,gradI)/sqrt(VectorLength(r)*VectorLength(gradI)));
                               }
                               w = dst*dir;
-                              Ia += (float)w * (float)(CV_MAT_ELEM(*out,data_type,k-1,l-1));
+                              Ia += (float)w * (float)(CV_MAT_ELEM(*out,data_type,km,lm));
                               s  += w;
                            }
                         }
@@ -704,18 +753,18 @@ icvInpaint( const CvArr* _input_img, const CvArr* _inpaint_mask, CvArr* _output_
     output_img = cvGetMat( _output_img, &output_hdr );
 
     if( !CV_ARE_SIZES_EQ(input_img,output_img) || !CV_ARE_SIZES_EQ(input_img,inpaint_mask))
-        CV_Error( cv::Error::StsUnmatchedSizes, "All the input and output images must have the same size" );
+        CV_Error( CV_StsUnmatchedSizes, "All the input and output images must have the same size" );
 
     if( (CV_MAT_TYPE(input_img->type) != CV_8U &&
          CV_MAT_TYPE(input_img->type) != CV_16U &&
          CV_MAT_TYPE(input_img->type) != CV_32F &&
         CV_MAT_TYPE(input_img->type) != CV_8UC3) ||
         !CV_ARE_TYPES_EQ(input_img,output_img) )
-        CV_Error( cv::Error::StsUnsupportedFormat,
+        CV_Error( CV_StsUnsupportedFormat,
         "8-bit, 16-bit unsigned or 32-bit float 1-channel and 8-bit 3-channel input/output images are supported" );
 
     if( CV_MAT_TYPE(inpaint_mask->type) != CV_8UC1 )
-        CV_Error( cv::Error::StsUnsupportedFormat, "The mask must be 8-bit 1-channel image" );
+        CV_Error( CV_StsUnsupportedFormat, "The mask must be 8-bit 1-channel image" );
 
     range = MAX(range,1);
     range = MIN(range,100);
@@ -737,6 +786,8 @@ icvInpaint( const CvArr* _input_img, const CvArr* _inpaint_mask, CvArr* _output_
     cvSet(t,cvScalar(1.0e6f,0,0,0));
     cv::dilate(cv::cvarrToMat(mask), cv::cvarrToMat(band), el_cross, cv::Point(1, 1));
     Heap=cv::makePtr<CvPriorityQueueFloat>();
+    if (!Heap->Init(band))
+        return;
     cvSub(band,mask,band,NULL);
     SET_BORDER1_C1(band,uchar,0);
     if (!Heap->Add(band))
@@ -752,6 +803,8 @@ icvInpaint( const CvArr* _input_img, const CvArr* _inpaint_mask, CvArr* _output_
         cv::dilate(cv::cvarrToMat(mask), cv::cvarrToMat(out), el_range);
         cvSub(out,mask,out,NULL);
         Out=cv::makePtr<CvPriorityQueueFloat>();
+        if (!Out->Init(out))
+            return;
         if (!Out->Add(band))
             return;
         cvSub(out,band,out,NULL);
