diff --git a/modules/videoio/src/cap_avfoundation.mm b/modules/videoio/src/cap_avfoundation.mm
index 2df990392f..12fa42f5b5 100644
--- a/modules/videoio/src/cap_avfoundation.mm
+++ b/modules/videoio/src/cap_avfoundation.mm
@@ -61,8 +61,6 @@
 
 #define DISABLE_AUTO_RESTART 999
 
-#if !TARGET_OS_VISION
-
 @interface CaptureDelegate : NSObject <AVCaptureVideoDataOutputSampleBufferDelegate>
 {
     int newFrame;
@@ -127,7 +125,6 @@ - (IplImage*)getOutput;
         int disableAutoRestart;
 };
 
-#endif
 
 /*****************************************************************************
  *
@@ -163,7 +160,6 @@ - (IplImage*)getOutput;
     uint32_t  mMode;
     int       mFormat;
 
-    void handleTracks(NSArray<AVAssetTrack *>* tracks, const char* filename);
     bool setupReadingAt(CMTime position);
     IplImage* retrieveFramePixelBuffer();
     int getPreferredOrientationDegrees() const;
@@ -221,19 +217,15 @@ - (IplImage*)getOutput;
 
 }
 
-
 cv::Ptr<cv::IVideoCapture> cv::create_AVFoundation_capture_cam(int index)
 {
-#if !TARGET_OS_VISION
     CvCaptureCAM* retval = new CvCaptureCAM(index);
     if (retval->didStart())
         return cv::makePtr<cv::LegacyCapture>(retval);
     delete retval;
-#endif
     return 0;
 }
 
-
 cv::Ptr<cv::IVideoWriter> cv::create_AVFoundation_writer(const std::string& filename, int fourcc,
                                                          double fps, const cv::Size &frameSize,
                                                          const cv::VideoWriterParameters& params)
@@ -253,8 +245,6 @@ - (IplImage*)getOutput;
  *
  *****************************************************************************/
 
-#if !TARGET_OS_VISION
-
 CvCaptureCAM::CvCaptureCAM(int cameraNum) {
     mCaptureSession = nil;
     mCaptureDeviceInput = nil;
@@ -783,7 +773,6 @@ -(int) updateImage {
 
 @end
 
-#endif
 
 /*****************************************************************************
  *
@@ -822,26 +811,24 @@ -(int) updateImage {
         return;
     }
 
-// Available since iOS 15
-#if TARGET_OS_VISION || (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 150000)
-    if (@available(iOS 15, *)) {
-        [mAsset loadTracksWithMediaType:AVMediaTypeVideo completionHandler:^(NSArray<AVAssetTrack *>* tracks, NSError* err) {
-            if (err != nil) {
-                handleTracks(tracks, filename);
-            }
-            [localpool drain];
-        }];
+    NSArray *tracks = [mAsset tracksWithMediaType:AVMediaTypeVideo];
+    if ([tracks count] == 0) {
+        fprintf(stderr, "OpenCV: Couldn't read video stream from file \"%s\"\n", filename);
+        [localpool drain];
+        started = 0;
         return;
-    } else {
-#if !TARGET_OS_VISION
-        NSArray *tracks = [mAsset tracksWithMediaType:AVMediaTypeVideo];
-        handleTracks(tracks, filename);
-#endif
     }
-#else
-    NSArray *tracks = [mAsset tracksWithMediaType:AVMediaTypeVideo];
-    handleTracks(tracks, filename);
-#endif
+
+    mAssetTrack = [tracks[0] retain];
+
+    if ( ! setupReadingAt(kCMTimeZero) ) {
+        fprintf(stderr, "OpenCV: Couldn't read movie file \"%s\"\n", filename);
+        [localpool drain];
+        started = 0;
+        return;
+    }
+
+    started = 1;
     [localpool drain];
 }
 
@@ -863,24 +850,6 @@ -(int) updateImage {
     [localpool drain];
 }
 
-void CvCaptureFile::handleTracks(NSArray<AVAssetTrack *>* tracks, const char* filename) {
-    if ([tracks count] == 0) {
-        fprintf(stderr, "OpenCV: Couldn't read video stream from file \"%s\"\n", filename);
-        started = 0;
-        return;
-    }
-
-    mAssetTrack = [tracks[0] retain];
-
-    if ( ! setupReadingAt(kCMTimeZero) ) {
-        fprintf(stderr, "OpenCV: Couldn't read movie file \"%s\"\n", filename);
-        started = 0;
-        return;
-    }
-
-    started = 1;
-}
-
 bool CvCaptureFile::setupReadingAt(CMTime position) {
     if (mAssetReader) {
         if (mAssetReader.status == AVAssetReaderStatusReading) {
@@ -1300,25 +1269,25 @@ -(int) updateImage {
         //exception;
     }
 
-    // Three codec supported AVVideoCodecTypeH264 AVVideoCodecTypeJPEG AVVideoCodecTypeHEVC
+    // Three codec supported AVVideoCodecH264 AVVideoCodecJPEG AVVideoCodecTypeHEVC
     // On iPhone 3G H264 is not supported.
     if (fourcc == CV_FOURCC('J','P','E','G') || fourcc == CV_FOURCC('j','p','e','g') ||
             fourcc == CV_FOURCC('M','J','P','G') || fourcc == CV_FOURCC('m','j','p','g')){
-        codec = [AVVideoCodecTypeJPEG copy]; // Use JPEG codec if specified, otherwise H264
+        codec = [AVVideoCodecJPEG copy]; // Use JPEG codec if specified, otherwise H264
     }else if(fourcc == CV_FOURCC('H','2','6','4') || fourcc == CV_FOURCC('a','v','c','1')){
-            codec = [AVVideoCodecTypeH264 copy];
+            codec = [AVVideoCodecH264 copy];
 // Available since iOS 11
-#if TARGET_OS_VISION || (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 110000)
+#if defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 110000
     }else if(fourcc == CV_FOURCC('H','2','6','5') || fourcc == CV_FOURCC('h','v','c','1') ||
             fourcc == CV_FOURCC('H','E','V','C') || fourcc == CV_FOURCC('h','e','v','c')){
         if (@available(iOS 11, *)) {
             codec = [AVVideoCodecTypeHEVC copy];
         } else {
-            codec = [AVVideoCodecTypeH264 copy];
+            codec = [AVVideoCodecH264 copy];
         }
 #endif
     }else{
-        codec = [AVVideoCodecTypeH264 copy]; // default canonical H264.
+        codec = [AVVideoCodecH264 copy]; // default canonical H264.
     }
 
     //NSLog(@"Path: %@", path);
@@ -1380,17 +1349,17 @@ -(int) updateImage {
     NSAutoreleasePool* localpool = [[NSAutoreleasePool alloc] init];
 
     [mMovieWriterInput markAsFinished];
-    [mMovieWriter finishWritingWithCompletionHandler:^() {
-        [mMovieWriter release];
-        [mMovieWriterInput release];
-        [mMovieWriterAdaptor release];
-        [path release];
-        [codec release];
-        [fileType release];
-        cvReleaseImage(&argbimage);
+    [mMovieWriter finishWriting];
+    [mMovieWriter release];
+    [mMovieWriterInput release];
+    [mMovieWriterAdaptor release];
+    [path release];
+    [codec release];
+    [fileType release];
+    cvReleaseImage(&argbimage);
+
+    [localpool drain];
 
-        [localpool drain];
-    }];
 }
 
 bool CvVideoWriter_AVFoundation::writeFrame(const IplImage* iplimage) {
