diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8802fb8..ace3ea0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,7 @@ target_include_directories(tinyxml2 PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SO
 
 target_compile_definitions(
     tinyxml2
-    PUBLIC $<$<CONFIG:Debug>:TINYXML2_DEBUG>
+    PUBLIC $<$<CONFIG:Debug>:TINYXML2_DEBUG> $<$<BOOL:${BUILD_SHARED_LIBS}>:TINYXML2_EXPORT>
     INTERFACE $<$<BOOL:${BUILD_SHARED_LIBS}>:TINYXML2_IMPORT>
     PRIVATE $<$<CXX_COMPILER_ID:MSVC>:_CRT_SECURE_NO_WARNINGS>
 )
@@ -36,7 +36,6 @@ target_compile_definitions(
 set_target_properties(
     tinyxml2
     PROPERTIES
-    DEFINE_SYMBOL "TINYXML2_EXPORT"
     VERSION "${tinyxml2_VERSION}"
     SOVERSION "${tinyxml2_VERSION_MAJOR}"
 )
diff --git a/tinyxml2.h b/tinyxml2.h
index 452ae95..22e45ef 100755
--- a/tinyxml2.h
+++ b/tinyxml2.h
@@ -72,7 +72,7 @@ distribution.
 #   else
 #       define TINYXML2_LIB
 #   endif
-#elif __GNUC__ >= 4
+#elif __GNUC__ >= 4 && defined TINYXML2_EXPORT
 #   define TINYXML2_LIB __attribute__((visibility("default")))
 #else
 #   define TINYXML2_LIB
