diff --git a/scripts/genout.cmake.in b/scripts/genout.cmake.in
index ec685d900..b673615a9 100644
--- a/scripts/genout.cmake.in
+++ b/scripts/genout.cmake.in
@@ -23,16 +23,6 @@ set(PNGLIB_VERSION "@PNGLIB_VERSION@")
 set(ZLIBINCDIR "@ZLIB_INCLUDE_DIRS@")
 
 set(PLATFORM_C_FLAGS)
-if(APPLE)
-  set(CMAKE_OSX_INTERNAL_ARCHITECTURES "@CMAKE_OSX_INTERNAL_ARCHITECTURES@")
-  set(CMAKE_OSX_SYSROOT "@CMAKE_OSX_SYSROOT@")
-  if(CMAKE_OSX_INTERNAL_ARCHITECTURES)
-    set(PLATFORM_C_FLAGS ${PLATFORM_C_FLAGS} -arch ${CMAKE_OSX_INTERNAL_ARCHITECTURES})
-  endif()
-  if(CMAKE_OSX_SYSROOT)
-    set(PLATFORM_C_FLAGS ${PLATFORM_C_FLAGS} -isysroot ${CMAKE_OSX_SYSROOT})
-  endif()
-endif()
 
 get_filename_component(INPUTEXT "${INPUT}" EXT)
 get_filename_component(OUTPUTEXT "${OUTPUT}" EXT)
